src/form/bf_mx_lin.jl

```diff
@@ -49,35 +65,53 @@ function constraint_mc_model_voltage_magnitude_difference(pm::LPUBFDiagModel, n:
     p_fr = var(pm, n, :p)[f_idx]
     q_fr = var(pm, n, :q)[f_idx]
 
-    p_s_fr = p_fr - diag(g_sh_fr).*w_fr
-    q_s_fr = q_fr + diag(b_sh_fr).*w_to
+    p_s_fr = p_fr- diag(g_sh_fr).*w_fr
+    q_s_fr = q_fr+ diag(b_sh_fr).*w_fr
+
+    fb = ref(pm, n, :bus, f_bus)
+    tb = ref(pm, n, :bus, t_bus)
+    
+    vmax1=fb["vmax"]
+    vmax2=tb["vmax"]
+    
+    cnds = conductor_ids(pm; nw=n)
+    ncnds = length(cnds)
 
     alpha = exp(-im*2*pi/3)
     Gamma = [1 alpha^2 alpha; alpha 1 alpha^2; alpha^2 alpha 1]
 
     MP = 2*(real(Gamma).*r + imag(Gamma).*x)
     MQ = 2*(real(Gamma).*x - imag(Gamma).*r)
 
-    JuMP.@constraint(pm.model, w_to .== w_fr - MP*p_s_fr - MQ*q_s_fr)
+    for c in 1:ncnds
+    if((vmax1[c]!=0) && (vmax2[c]!=0))
+      a=JuMP.@constraint(pm.model, w_to[c]*1000.0 == 1000.0*(w_fr[c] - (MP[c,1]*p_s_fr[1]+MP[c,2]*p_s_fr[2]+MP[c,3]*p_s_fr[3]) - (MQ[c,1]*q_s_fr[1]+MQ[c,2]*q_s_fr[2]+MQ[c,3]*q_s_fr[3])))
+    else
+      JuMP.@constraint(pm.model, p_s_fr[c]*1000.0==0)
+      JuMP.@constraint(pm.model, q_s_fr[c]*1000.0==0)
+    end
+   end
 
 end
 
 
+
 "balanced three-phase phasor"
 function constraint_mc_theta_ref(pm::LPUBFDiagModel, n::Int, i::Int, va_ref)
     ncnds = length(conductor_ids(pm))
     @assert(ncnds >= 2)
-
     w = var(pm, n, :w)[i]
-    JuMP.@constraint(pm.model, w[2:ncnds]   .== w[1])
+    #JuMP.@constraint(pm.model, w[2:ncnds]   .== w[1])
```

Comment:
inadvertent comment out?

Reply:
I am not fully sure that the constraint is correct., I think the voltage angles are meant to be equal as the name of the function suggests (constraint_mc_theta_ref), not the voltage magnitudes

Reply:
I think this is part of a discussion we have been having about reference bus constraints needing magnitudes to be fixed too, not just angles. I agree that this constraint is probably not in the right place though just based on the name of the function. Comments @frederikgeth or @sanderclaeys ?

Reply:
ideally we support these cases:
1. a bus where both the voltage magnitude and angles are fixed
2. a bus where the voltage angles are fixed, the voltage magnitudes are equal between the phases, but not set to a specific value

Either condition is sufficient to guarantee that a formulation in the lifted variable space is meaningful. I'm happy with supporting just case 1 though. We'll need to change the implementation of constraint_mc_theta_ref for the SDP relaxation as well though.

Reply:
@smithagopinath I think this should be changed back to being uncommented, and I will make a Github Issue that this should be addressed.